release/man/riprip.1
.TH "RIP RIP HOORAY!" "1" "October 2026" "riprip v0.9.1" "User Commands"
.SH NAME
RIP RIP HOORAY! \- Manual page for riprip v0.9.1.
.SH DESCRIPTION
A specialized audio CD\-ripper optimized for incremental data recovery.
.SS USAGE:
.TP
riprip [FLAGS] [OPTIONS]
.SS FLAGS:
.TP
\fB\-\-backwards\fR
Reverse the sector read order when ripping a track, starting at end, and ending at the start.
.TP
\fB\-\-flip\-flop\fR
Alternate the sector read order between passes, forwards then backwards then forwards then backwards… This has no effect unless \-p/\-\-passes is at least two.
.TP
\fB\-h\fR, \fB\-\-help\fR
Print help information to STDOUT and exit.
.TP
\fB\-\-no\-metadata\fR
Do not try to read/parse CD\-Text, MCN, and track ISRC metadata from the disc. This can speed up initialization, particularly for discs with a lot of damage.
.TP
\fB\-\-no\-resume\fR
Ignore any previous rip states, starting over from scratch.
.TP
\fB\-\-no\-rip\fR
Print the basic drive and disc information to STDERR and exit (without ripping anything).
.TP
\fB\-\-no\-summary\fR
Skip the drive and disc summary and jump straight to ripping.
.TP
\fB\-\-reset\fR
Flip 'likely' samples back to 'maybe', keeping their values, but resetting all counts to one. This is a softer alternative to \-\-no\-resume, and will not affect tracks confirmed by AccurateRip/CUETools.
.TP
\fB\-\-status\fR
Print the status of the individual track rips (that you presumably already started) to STDERR and exit. Note that only the \-\-no\-summary, \-\-confidence, and \-r/\-\-rereads options have any meaning in this mode.
.TP
\fB\-\-strict\fR
Consider C2 errors an all\-or\-nothing proposition for the sector as a whole, marking all samples bad if any of them are bad. This is most effective when applied consistently from the initial rip and onward.
.TP
\fB\-\-sync\fR
Confirm sector positioning with subchannel data (when available) to make sure the drive is actually reading from the right place, and ignore the data if not. This is prone to false\-positives — subchannel data is easily corrupted — so only recommended when disc rot, rather than wear\-and\-tear, is the sole cause of your woes.
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Print a detailed log of the proceedings — including which specific samples and sectors are having trouble — to STDOUT so you can redirect it to a location of your choosing, e.g. `riprip \-v > out.log`. May be specified up to four times for more detail.
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version information to STDOUT and exit.
.SS OPTIONS:
.TP
\fB\-c\fR, \fB\-\-cache\fR <NUM>
Drive cache can interfere with re\-read accuracy. If your drive caches data, use this option to specify its buffer size so Rip Rip can try to mitigate it. Values with an M suffix are treated as MiB, otherwise KiB are assumed. [default: auto or 0; max: 65,535]
.TP
\fB\-\-confidence\fR <NUM>
Consider a track accurately ripped — i.e. stop working on it — AccurateRip and/or CUETools matches are found with a confidence of at least <NUM>. [default: 3; range: 1..=10]
.TP
\fB\-d\fR, \fB\-\-dev\fR <PATH>
The device path for the optical drive containing the CD of interest, like /dev/cdrom. [default: auto]
.TP
\fB\-o\fR, \fB\-\-offset\fR <SAMPLES>
The AccurateRip, et al, sample read offset to apply to data retrieved from the drive. [default: auto or 0; range: ±5880]
.TP
\fB\-p\fR, \fB\-\-passes\fR <NUM>
Automate re\-ripping by executing up to <NUM> passes for each track while any samples remain unread or unconfirmed. [default: 1; max: 16]
.TP
\fB\-r\fR, \fB\-\-rereads\fR <[ABS],[MUL]>
Re\-read sectors on subsequent passes until A) they have been independently verified with AccurateRip or CUETools; or B) the same allegedly\-good values have been read at least <ABS> times, and <MUL> times more often than any contradictory 'good' values. The value may omit the number on either side of the comma to keep the default, or be a single number to alter only the <ABS>. [default: 2,2; range: 1..=20,1..=10]
.TP
\fB\-t\fR, \fB\-\-tracks\fR <NUM>
Rip one or more specific tracks (rather than the whole disc). Multiple tracks can be separated by commas (2,3), specified as an inclusive range (2\-3), and/or given their own \-t/\-\-track (\-t 2 \-t 3). Track 0 can be used to rip the HTOA, if any. [default: the whole disc]


release/completions/riprip.bash
_basher___riprip() {
	local cur prev opts
	COMPREPLY=()
	cur="${COMP_WORDS[COMP_CWORD]}"
	prev="${COMP_WORDS[COMP_CWORD-1]}"
	opts=()
	[[ " ${COMP_LINE} " =~ " --backwards " ]] || opts+=("--backwards")
	[[ " ${COMP_LINE} " =~ " --flip-flop " ]] || opts+=("--flip-flop")
	if [[ ! " ${COMP_LINE} " =~ " -h " ]] && [[ ! " ${COMP_LINE} " =~ " --help " ]]; then
		opts+=("-h")
		opts+=("--help")
	fi
	[[ " ${COMP_LINE} " =~ " --no-metadata " ]] || opts+=("--no-metadata")
	[[ " ${COMP_LINE} " =~ " --no-resume " ]] || opts+=("--no-resume")
	[[ " ${COMP_LINE} " =~ " --no-rip " ]] || opts+=("--no-rip")
	[[ " ${COMP_LINE} " =~ " --no-summary " ]] || opts+=("--no-summary")
	[[ " ${COMP_LINE} " =~ " --reset " ]] || opts+=("--reset")
	[[ " ${COMP_LINE} " =~ " --status " ]] || opts+=("--status")
	[[ " ${COMP_LINE} " =~ " --strict " ]] || opts+=("--strict")
	[[ " ${COMP_LINE} " =~ " --sync " ]] || opts+=("--sync")
	opts+=("-v")
	opts+=("--verbose")
	if [[ ! " ${COMP_LINE} " =~ " -V " ]] && [[ ! " ${COMP_LINE} " =~ " --version " ]]; then
		opts+=("-V")
		opts+=("--version")
	fi
	if [[ ! " ${COMP_LINE} " =~ " -c " ]] && [[ ! " ${COMP_LINE} " =~ " --cache " ]]; then
		opts+=("-c")
		opts+=("--cache")
	fi
	[[ " ${COMP_LINE} " =~ " --confidence " ]] || opts+=("--confidence")
	if [[ ! " ${COMP_LINE} " =~ " -d " ]] && [[ ! " ${COMP_LINE} " =~ " --dev " ]]; then
		opts+=("-d")
		opts+=("--dev")
	fi
	if [[ ! " ${COMP_LINE} " =~ " -o " ]] && [[ ! " ${COMP_LINE} " =~ " --offset " ]]; then
		opts+=("-o")
		opts+=("--offset")
	fi
	if [[ ! " ${COMP_LINE} " =~ " -p " ]] && [[ ! " ${COMP_LINE} " =~ " --passes " ]]; then
		opts+=("-p")
		opts+=("--passes")
	fi
	if [[ ! " ${COMP_LINE} " =~ " -r " ]] && [[ ! " ${COMP_LINE} " =~ " --rereads " ]]; then
		opts+=("-r")
		opts+=("--rereads")
	fi
	opts+=("-t")
	opts+=("--tracks")
	opts=" ${opts[@]} "
	if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
		COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
		return 0
	fi
	case "${prev}" in
		--dev|-d)
			if [ -z "$( declare -f _filedir )" ]; then
				COMPREPLY=( $( compgen -f "${cur}" ) )
			else
				COMPREPLY=( $( _filedir ) )
			fi
			return 0
			;;
		*)
			COMPREPLY=()
			;;
	esac
	COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
	return 0
}
complete -F _basher___riprip -o bashdefault -o default riprip
